draft/dc_config.txt

```diff
@@ -0,0 +1,150 @@
+
+This is a one-time check, but could be automated with a Python script
+that connect-scans all the servers, tries common ports and checks
+their certificate.
+
+4. strict TLS should be applied even if no servers are defined.
+
+I DeltaChat: src/configure/mod.rs:433: offline autoconfig found, but no servers defined
+
+After this error, autoconfig is not applied at all, even though the
+database has "strict_tls: true".
+
+This happens for dubby.org, which has no server listed in the
+provider database now.
+
+5. config: split IMAP and SMTP server flags.
+
+There is an issue for that:
+https://github.com/deltachat/deltachat-core-rust/issues/646
+
+It can be done easier than listed in an issue, with a database
+migration and breaking change, deprecating security flags stored in
+server_flags immediately.
+
+6. configure: untangle IMAP and SMTP configuration
+
+Currently both configurations are stored in LoginParam structure.
+
+After moving IMAP and SMTP security flags into a separate enum, the
+structure can be split into independent IMAP and SMTP configuration.
+
+7. configure: attempt IMAP and SMTP configuration in parallel
+
+It should be done in separate async tasks to speed up configuration,
+especially if multiple servers should be tried.
+
+8. configure: fix the error displayed in case of failed configuration
+
+Configuration process returns the last error if configuration of IMAP
+or SMTP fails.
+
+If no provider database entry is used and autoconfiguration is not
+available or can't be downloaded, the last error is about the last
+domain name tried, which may not exist at all.
+
+For example:
+
+E DeltaChat: Response from IMAP: mail.example.org:993: io: could not resolve address `("mail.example.org", 993)
+
+It is confusing, because users then try to figure out why connection
+to mail.example.org does not work for them, while IMAP server is
+actually example.org, but connection to it failed for other reasons,
+like incorrect login or blocked port.
+
+The error should list all the servers tried and the outcome for it.
+Maybe transmit each outcome in a separate event for UI to display in a
+"details" section below the progress bar:
+
+Trying example.org ... port closed
+Trying imap.example.org ... could not resolve address
+Trying mail.example.org ... incorrect login
+
+9. configure: connection refused error suggests to check the inbox, even if IMAP can't connect
+
+If connection is refused, it does not make sense to suggest checking
+the inbox. Especially if IMAP failed or domain can't be resolved.
+
+Current error message is too broad and usually irrelevant and not helpful:
+
+Some providers place additional information in your inbox; you can
+check them eg. in the web frontend. Consult your provider or friends
+if you run into problems.
+
+Which providers place information about failed login attempts into
+inbox? Is it helpful or just "someone from ... tried to login with
+incorrect password"?
```

Comment:
GMX and Yahoo do send you a helpful error email IIRC. However, the provider-db also shows this information, so it is actually not needed. 

Moreover, many users will probably not understand "Some providers place additional information in your inbox; you can check them eg. in the web frontend." And "Consult your provider or friends if you run into problems" is a nice advice but many users will ask their techie friends anyway and the provider will probably not help.

Together with point 8 we could do:
IMAP login failed.
Trying example.org ... port closed
Trying imap.example.org ... could not resolve address
Trying mail.example.org ... incorrect login

(and if it seems to be the password, tell the user to check the pw, as we already do)